File: lib/src/modals/customer.dart
class Customer {
  final String name;
  double outstandingAmount;
  String paymentStatus;
  String dueDate;
  int dueMonth;
  double dueAmount;

  Customer({
    required this.name,
    required this.outstandingAmount,
    required this.paymentStatus,
    required this.dueDate,
    required this.dueMonth,
    required this.dueAmount,
  });

  factory Customer.fromJson(Map<String, dynamic> json) {
    return Customer(
      name: json['name'] ?? 'Unknown',
      outstandingAmount: (json['amount'] ?? 0.0).toDouble(),
      paymentStatus: json['paymentStatus'] ?? 'Pending',
      dueDate: json['dueDate'] ?? 'N/A',
      dueMonth: (json['dueMonth'] ?? 0).toInt(),
      dueAmount: (json['paidAmount'] ?? 0.0).toDouble(),
    );
  }


}

File: payment-system-complete/models/customer.js

const mongoose = require('mongoose');

const customerSchema = new mongoose.Schema({
    refNumber: { type: String, required: true },
    name: { type: String, required: true },
    amount: { type: Number, required: true },
    dueDate: { type: String, required: true },
    paidAmount: { type: Number, required: true },
    dueMonth: { type: Number, required: true }
});

module.exports = mongoose.model('Customer', customerSchema);
